### DISCUSSION TITLE
is there an example that shows how this would work with echo or chi ?

### DISCUSSION THREAD

**Discussion by ymolists** [ogen-go/ogen#1196] [Q&A]
I am writing both the server and client using ogen.  i am trying to test the generated code using unit / integ tests. can you please point me to examples that show how one would

1) integrate the generated code with echo / chi
2) unit / integ tests the generated code with echo / chi

right now i am creating an instance of a handler and calling functions directly. Which is not testing the http layer errors. how do people unit test the generated code while including the http layer ?

```go
                reqServer := ...
		stages, err = reqServer.CliCreateExecuteStage(context.Background(), &req00)
		if err != nil {
			t2.Fatal("client error:", err)
		}

```

**Comment by akkuman** [comment]
## chi
https://ogen.dev/docs/concepts/static_router#adding-profiler-metrics-or-static-content-route

## gin

```go
srv, err := oas.NewServer(h)
if err != nil {
      log.Fatal(err)
  }
r := gin.Default()
r.Any("/api/v1/*any", gin.WrapH(http.StripPrefix("/api/v1", srv)))
r.Run()
```
